samcli/commands/deploy/guided_context.py

```diff
@@ -177,17 +181,30 @@ def guided_prompts(self, parameter_override_keys):
             )
 
         click.echo("\n\tLooking for resources needed for deployment:")
-        s3_bucket = manage_stack(profile=self.profile, region=region)
-        click.secho(f"\n\tManaged S3 bucket: {s3_bucket}", bold=True)
-        click.echo("\tA different default S3 bucket can be set in samconfig.toml")
+        managed_s3_bucket = manage_stack(profile=self.profile, region=region)
+        click.secho(f"\n\tManaged S3 bucket: {managed_s3_bucket}", bold=True)
+        click.echo(
+            "\tA different default S3 bucket can be set in samconfig.toml"
+            " and auto resolution of buckets turned off by setting resolve_s3=False"
+        )
 
-        image_repositories = self.prompt_image_repository(
-            stack_name, stacks, self.image_repositories, region, s3_bucket, self.s3_prefix
+        image_repositories = (
+            sync_ecr_stack(
+                self.template_file, stack_name, region, managed_s3_bucket, self.s3_prefix, self.image_repositories
+            )
+            if self.resolve_image_repositories
+            else self.prompt_image_repository(
+                stack_name, stacks, self.image_repositories, region, managed_s3_bucket, self.s3_prefix
+            )
         )
 
         self.guided_stack_name = stack_name
-        self.guided_s3_bucket = s3_bucket
+        self.guided_s3_bucket = managed_s3_bucket
         self.guided_image_repositories = image_repositories
+        # NOTE(sriram-mv): The resultant s3 bucket is ALWAYS the managed_s3_bucket. There is no user flow to set it
+        # within guided.
+        self.resolve_s3 = True if self.guided_s3_bucket else False
```

Comment:
Very much a nit comment and might be more on the side of personal preference 😄 
```suggestion
        self.resolve_s3 = bool(self.guided_s3_bucket)
```